### PR TITLE
Fix Numeric strings being converted to Datetime objects

### PR DESCRIPTION
**Pull request by shemogumbe**
Fixes https://github.com/microsoftgraph/msgraph-sdk-python/issues/653

## Overview

The PR fixes a problem where pendulum assumes all 4-digit numeric strings are Datetimes, e. 1212, 1214, 2530, 9057

Fixes # https://github.com/microsoftgraph/msgraph-sdk-python/issues/653

### The Script below,
```python
async def get_my_messages():
    try:

        result = await user_client.sites.by_site_id(
            "yygbp.sharepoint.com,ab839da2-a22d-4e5c-bbeb-5060fad36e50,5cfc6d41-e5f6-4341-93fb-98728649efde"
        ).lists.by_list_id('6c8b411a-e440-4b24-9ee3-bfd3f06a9b4e'
                           ).items.by_list_item_id('1').get()
        print(f"sharepont list item id {result.id}")
        print(f"sharepoint list item fields {result.fields}")
        print(
            f"sharepoint list item text field {result.fields.additional_data['Doc_id']}"
        )
```
###  Output before
```
sharepoint list item fields FieldValueSet(additional_data={'@odata.etag': '"0f11e7ca-3f0e-46ce-b7e7-f2dfffe373da,2"', 'Title': 'Composition', 'LinkTitle': 'Composition', 'Doc_id': DateTime(1212, 1, 1, 0, 0, 0, tzinfo=Timezone('UTC')), 'Doc_numner': 1212.0, 'ContentType': 'Item', 'Modified': DateTime(2024, 8, 23, 14, 54, 17, tzinfo=Timezone('UTC')), 'Created': DateTime(2024, 8, 23, 14, 54, 17, tzinfo=Timezone('UTC')), 'AuthorLookupId': '9', 'EditorLookupId': '9', '_UIVersionString': '2.0', 'Attachments': True, 'Edit': '', 'LinkTitleNoMenu': 'Composition', 'ItemChildCount': '0', 'FolderChildCount': '0', '_ComplianceFlags': '', '_ComplianceTag': '', '_ComplianceTagWrittenTime': '', '_ComplianceTagUserId': ''}, id='1', odata_type=None)
sharepoint list item text field 1212-01-01 00:00:00+00:00

```
### Output After 
```
sharepoint list item fields FieldValueSet(additional_data={'@odata.etag': '"0f11e7ca-3f0e-46ce-b7e7-f2dfffe373da,2"', 'Title': 'Composition', 'LinkTitle': 'Composition', 'Doc_id': '1212', 'Doc_numner': 1212.0, 'ContentType': 'Item', 'Modified': DateTime(2024, 8, 23, 14, 54, 17, tzinfo=Timezone('UTC')), 'Created': DateTime(2024, 8, 23, 14, 54, 17, tzinfo=Timezone('UTC')), 'AuthorLookupId': '9', 'EditorLookupId': '9', '_UIVersionString': '2.0', 'Attachments': True, 'Edit': '', 'LinkTitleNoMenu': 'Composition', 'ItemChildCount': '0', 'FolderChildCount': '0', '_ComplianceFlags': '', '_ComplianceTag': '', '_ComplianceTagWrittenTime': '', '_ComplianceTagUserId': ''}, id='1', odata_type=None)
sharepoint list item text field 1212
```
### Notes

This change could be reverted  if pendulum does an update that fixes the bug.

### CODE REVIEW

File: kiota_serialization_json/json_parse_node.py [hunk 1]
```diff
@@ -291,6 +293,10 @@ def _assign_field_values(self, item: U) -> None:
                     deserialize but the model doesn't support additional data"
                 )
 
+    def is_four_digit_number(self, value: str) -> bool:
```
Comment (baywet) [inline kiota_serialization_json/json_parse_node.py:296]: just making sure this method is not accessible from outside of the class?

File: kiota_serialization_json/json_parse_node.py [hunk 2]
```diff
@@ -300,11 +306,15 @@ def try_get_anything(self, value: Any) -> Any:
             return dict(map(lambda x: (x[0], self.try_get_anything(x[1])), value.items()))
         if isinstance(value, str):
             try:
+                if self.is_four_digit_number(value):
+                    print(value)
```
Comment (baywet) [inline kiota_serialization_json/json_parse_node.py:310]:
```suggestion
```

File: kiota_serialization_json/json_parse_node.py [hunk 3]
```diff
@@ -300,11 +306,15 @@ def try_get_anything(self, value: Any) -> Any:
...
+                    return value
```
Comment (baywet) [inline kiota_serialization_json/json_parse_node.py:311]: does the value need to be casted? or parse to an integer?

**Review by baywet — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Also:
- unit tests
- changelog
- version bump
- replicate in other serialization libraries

File: kiota_serialization_json/json_parse_node.py (hunk 1 above)
Comment (shemogumbe) [reply kiota_serialization_json/json_parse_node.py:296]: It's an instance method, and we really just need it here in the entire project, so nope, not accessible outside the class

File: kiota_serialization_json/json_parse_node.py (hunk 3 above)
Comment (shemogumbe) [reply kiota_serialization_json/json_parse_node.py:310]: At this point, the value is already a json string, numeric types are dealt with in the get_int_value and get_float_value methods

File: kiota_serialization_json/json_parse_node.py (hunk 1 above)
Comment (baywet) [reply kiota_serialization_json/json_parse_node.py:296]: shouldn't it start with __ ?

File: kiota_serialization_json/_version.py [hunk 4]
```diff
@@ -1 +1 @@
-VERSION: str = '1.3.0'
+VERSION: str = '1.4.0'
```
Comment (baywet) [inline kiota_serialization_json/_version.py:1]: nit: a patch bump would have been sufficient according to the semver conventions.

**Review by baywet — APPROVED** [review]
Thank you for making the changes!

**Comment by shemogumbe** [comment]
> Thanks for the contribution! Also:
> 
> * unit tests
> * changelog
> * version bump
> * replicate in other serialization libraries

Added Unit tests, changelog and version bump

The other serialization libraries do not use pendulum thus do not have the problem

**Review by baywet — APPROVED** [review]

**Pull request merged**